erpnext/public/js/utils/serial_batch_selector.js

```diff
@@ -102,6 +118,25 @@ erpnext.stock.SerialBatchSelector = Class.extend({
 		this.dialog.show();
 	},
 
+	load_all_batches: function () {
+		frappe.call({
+			method: "erpnext.stock.doctype.batch.batch.get_sufficient_batch_or_fifo",
+			args: {
+				qty: this.doc.qty,
```

Comment:
This seems like it will always select batches as per FIFO. But we have 2 cases.
FIFO Button: select as per FIFO (no problem here)
Batch Selector Setup and Warehouse Change event: select as per already selected in transaction items